### PR TITLE
add ingress support to fullnode helm chart

### PR DESCRIPTION
**Pull request by PringBlocks**
### Description
For operators that are running a kubernetes cluster, it is common to have many services exposed publicly via ingress resources. This PR enables these operators to leverage their existing ingress controller for fullnode deployments.

### Test Plan
I have these changes deployed to my cluster already and everything is working as expected

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2651)

### CODE REVIEW

File: terraform/helm/fullnode/values.yaml [hunk 1]
```diff
@@ -44,6 +44,14 @@ service:
   loadBalancerSourceRanges: []
   annotations: {}
 
+ingress:
+  #change enabled to true and fill out the rest of the fields to expose the REST API externally via your ingress controller
+  enabled: false
+  hostName: #aptos-fullnode.example.com
+  #leaving class empty will result in an ingress that implicity uses the default ingress class
+  class: #nginx
```
Comment (sherry-x) [inline terraform/helm/fullnode/values.yaml:52]: nit: could you please update this to `ingressClassName ` instead?

File: terraform/helm/fullnode/templates/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{{- if .Values.ingress.enabled -}}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ include "aptos-fullnode.fullname" . }}-ingress
```
Comment (sherry-x) [inline terraform/helm/fullnode/templates/ingress.yaml:5]: I'd add `annotations` and labels for metadata as well, see example https://github.com/aptos-labs/aptos-core/blob/main/terraform/helm/fullnode/templates/service.yaml#L5-L10

**Review by sherry-x — COMMENTED** [review]
Thanks for submitting the PR, overall looks good, just couple small feedbacks

**Comment by PringBlocks** [comment]
thanks for taking the time to review my PR! I just pushed a commit with your requested changes

**Review by sherry-x — APPROVED** [review]

**Pull request merged**